analysis/topEFT/topeft.py

```diff
@@ -156,163 +149,163 @@ def process(self, events):
+
+          #################### Jets ####################
+
+          # Jet cleaning, before any jet selection
+          #vetos_tocleanjets = ak.with_name( ak.concatenate([tau, l_fo], axis=1), "PtEtaPhiMCandidate")
+          vetos_tocleanjets = ak.with_name( l_fo, "PtEtaPhiMCandidate")
+          tmp = ak.cartesian([ak.local_index(jets.pt), vetos_tocleanjets.jetIdx], nested=True)
+          cleanedJets = jets[~ak.any(tmp.slot0 == tmp.slot1, axis=-1)] # this line should go before *any selection*, otherwise lep.jetIdx is not aligned with the jet index
+
+          # Selecting jets and cleaning them
+          jetptname = "pt_nom" if hasattr(cleanedJets, "pt_nom") else "pt"
+
+          # Jet energy corrections
+          if not isData:
             cleanedJets["pt_raw"] = (1 - cleanedJets.rawFactor)*cleanedJets.pt
             cleanedJets["mass_raw"] = (1 - cleanedJets.rawFactor)*cleanedJets.mass
-            cleanedJets["pt_gen"] = ak.values_astype(ak.fill_none(cleanedJets.matched_gen.pt, 0), np.float32)
+            cleanedJets["pt_gen"] =ak.values_astype(ak.fill_none(cleanedJets.matched_gen.pt, 0), np.float32)
             cleanedJets["rho"] = ak.broadcast_arrays(events.fixedGridRhoFastjetAll, cleanedJets.pt)[0]
             events_cache = events.caches[0]
-            corrected_jets = ApplyJetCorrections.build(cleanedJets, lazy_cache=events_cache)
-            '''
+            cleanedJets = ApplyJetCorrections(year, corr_type='jets').build(cleanedJets, lazy_cache=events_cache)
             # SYSTEMATICS
-            jets = corrected_jets
-            if(self.jetSyst == 'JERUp'):
-                jets = corrected_jets.JER.up
-            elif(self.jetSyst == 'JERDown'):
-                jets = corrected_jets.JER.down
-            elif(self.jetSyst == 'JESUp'):
-                jets = corrected_jets.JES_jes.up
-            elif(self.jetSyst == 'JESDown'):
-                jets = corrected_jets.JES_jes.down
-            '''
-
-        cleanedJets["isGood"] = isTightJet(getattr(cleanedJets, jetptname), cleanedJets.eta, cleanedJets.jetId, jetPtCut=30.) # temporary at 25 for synch, TODO: Do we want 30 or 25?
-        goodJets = cleanedJets[cleanedJets.isGood]
-
-        # Count jets
-        njets = ak.num(goodJets)
-        ht = ak.sum(goodJets.pt,axis=-1)
-        j0 = goodJets[ak.argmax(goodJets.pt,axis=-1,keepdims=True)]
-
-        # Loose DeepJet WP
-        # TODO: Update these numbers when UL16 is available, and double check UL17 and UL18 at that time as well
-        if year == "2017":
+            if(syst_var == 'JERUp'): cleanedJets = cleanedJets.JER.up
+            elif(syst_var == 'JERDown'): cleanedJets = cleanedJets.JER.down
+            elif(syst_var == 'JESUp'): cleanedJets = cleanedJets.JES_jes.up
+            elif(syst_var == 'JESDown'): cleanedJets = cleanedJets.JES_jes.down
+            met=ApplyJetCorrections(year, corr_type='met').build(met_raw, cleanedJets, lazy_cache=events_cache)
```

Comment:
It might be cleaner to create a function that takes `syst_var` and spits out the correct `cleanedJets` collection. I won't push if you feel it is unnecessary.

Reply:
Ok, done!

Reply:
Thanks!